figures/views.py

```diff
@@ -453,6 +453,19 @@ def retrieve(self, request, *args, **kwargs):
         course_overview = get_object_or_404(CourseOverview, pk=course_key)
         return Response(CourseDetailsSerializer(course_overview).data)
 
+    def get_serializer_context(self):
+        context = super(CourseDetailsViewSet, self).get_serializer_context()
+        start_date = self.request.GET.get('start_date')
+        end_date = self.request.GET.get('end_date')
+        context.update({
+            "request": self.request,
+            "site": self.request.site,
```

Comment:
why are we passing `request` and `site` objects?

Reply:
In case of site..  They were using self.site in serialzer.  the serializer self object does not have site linked to it to it always returned None. 
That is why in order to pass site in get_course_history_metric() we need to bring it from views in context.

Reply:
wasn't `self.site` working fine before?